librecad/src/lib/information/rs_information.cpp

```diff
@@ -109,12 +110,12 @@ bool RS_Information::isTrimmable(RS_Entity* e1, RS_Entity* e2) {
                     e1->getParent()==e2->getParent()) {
 
                 // in the same polyline
-                RS_EntityContainer* pl = e1->getParent();
+                RS_Polyline* pl = (RS_Polyline *)e1->getParent();
                 int idx1 = pl->findEntity(e1);
                 int idx2 = pl->findEntity(e2);
                 RS_DEBUG->print("RS_Information::isTrimmable: "
                                 "idx1: %d, idx2: %d", idx1, idx2);
-                if (abs(idx1-idx2)==1 || abs(idx1-idx2)==int(pl->count()-1)) {
+                if (abs(idx1-idx2)==1 || (pl->isClosed() && abs(idx1-idx2)==int(pl->count()-1))) {
```

Comment:
looks like it's an API issue by design. The API requires that neighbors have consecutive indices and are actually connected by end/start points.

Probably too difficult to change this.

Ideally, we should provide an iterator here to guarantee looping through neighbors.